lib/slack/web/api/mixins/channels.id.rb

```diff
@@ -17,7 +17,7 @@ def channels_id(options = {})
             throw ArgumentError.new('Required arguments :channel missing') if name.nil?
 
             id_for(:channel, name, '#', :channels, 'channel_not_found') do
-              channels_list
+              conversations_list
```

Comment:
When users call `channels_` they get deprecations. I think we want to keep that. For those that call `conversations_` we shouldn't have any deprecations. I don't think we should mix both. It's in `id_for` that we want to call the new(er) methods.

Reply:
I didn't quite follow this
> It's in id_for that we want to call the new(er) methods

If I understood correctly, the `conversations_`-methods still use this same `channels_id` method, which would result in calling the deprecated `channels_list` method. Did you mean we should have some logic to determine whether `channels_list` or `conversations_list` is used, depending on whether it's a `channels_` or `conversations_` method call?

If so, I could make a copy of this method that uses `conversations_list`, and in the method templates use `deprecated: true` as a signal for whether to use this new method instead of `channels_id`. What do you think?

Reply:
I'll try again.

- introduce `Mixins::Conversations::Ids#conversations_id` that looks just like this method except change channels to conversations
- deprecate `channels_id` the usual way with a warning
- call `conversations_id` everywhere we call `channels_id` except in deprecated `channels_list` method

This way we don't change this mix-in and fix any method that has not been deprecated.

Reply:
Sounds good, will go with that!